symfony/client/src/orangehrmAdminPlugin/pages/payGrade/AddPayGrade.vue

```diff
@@ -0,0 +1,126 @@
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6" class="orangehrm-main-title">
+        Add Pay Grade
+      </oxd-text>
+
+      <oxd-divider />
+
+      <oxd-form :loading="isLoading" @submitValid="onSave">
+        <oxd-form-row>
+          <oxd-input-field
+            label="Name"
+            v-model="grade.name"
+            :rules="rules.name"
+            required
+          />
+        </oxd-form-row>
+
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button
+            type="button"
+            displayType="ghost"
+            label="Cancel"
+            @click="onCancel"
+          />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {navigate} from '@orangehrm/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+import {required} from '@orangehrm/core/util/validation/rules';
+
+export default {
+  data() {
+    return {
+      isLoading: false,
+      grade: {
+        id: '',
+        name: '',
+      },
+      rules: {
+        name: [],
+      },
+      errors: [],
+    };
+  },
+
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      'api/v2/admin/pay-grades',
+    );
+    return {
+      http,
+    };
+  },
+
+  methods: {
+    onSave() {
+      this.isLoading = true;
+      this.http
+        .create({
+          name: this.grade.name,
+        })
+        .then(response => {
+          const {data} = response.data;
+          this.$toast.addSuccess();
+          navigate('/admin/editPayGrade/{id}', {id: data.id});
+        });
+    },
+    onCancel() {
+      navigate('/admin/viewPayGrades');
+    },
+  },
+
+  created() {
+    this.isLoading = true;
+    this.http
+      .getAll()
+      .then(response => {
+        const {data} = response.data;
+        this.rules.name.push(required);
+        this.rules.name.push(v => {
+          return (v && v.length <= 50) || 'Should not exceed 50 characters';
```

Comment:
`shouldNotExceedCharLength()` function is now available in `@orangehrm/core/util/validation/rules` can be a drop in replacement.

Reply:
Done